sweep/tx_input_set.go

```diff
@@ -108,19 +123,33 @@ func (t *txInputSet) add(input input.Input, fromWallet bool) bool {
 	// added to the set.
 	newOutputValue := newInputTotal - fee
 
-	// If adding this input makes the total output value of the set
-	// decrease, this is a negative yield input. We don't add the input to
-	// the set and return the outcome.
-	if newOutputValue <= t.outputValue {
-		return false
-	}
+	// Initialize new wallet total with the current wallet total. This is
+	// updated below if this input is a wallet input.
+	newWalletTotal := t.walletInputTotal
```

Comment:
seems like the handling of this field can stay inside the switch case

Reply:
I moved it outside, because otherwise you have to be very careful not to update `t.walletInputTotal` before you do any additional checks. In that case, the total would be updated without the add happening.